src/Amalgam/GeneralizedDistance.h

```diff
@@ -389,6 +414,34 @@ class GeneralizedDistance
 			return fastPowP.FastPow(d);
 	}
 
+	//exponentiats and weights the difference term contextually based on pValue
+	//note that it has extra logic to account for extreme values like infinity, negative infinity, and 0
+	__forceinline double ContextuallyExponentiateAndWeightDifferenceTerm(double dist_term, size_t index, bool high_accuracy)
+	{
+		if(dist_term == 0.0)
+			return 0.0;
+
+		double weight = featureParams[index].weight;
```

Comment:
Do you need to verify index is valid here? Or precondition that it should always be in range?

Reply:
Since it is basically everywhere, I don't think there is a good place for it. So I'll resolve this.